<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Huwenbo Shi">
        <link rel="canonical" href="https://huwenboshi.github.io/pesca/posterior/">
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Estimating per-SNP posteriors - PESCA</title>
        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-XXXXXXXX-X', 'auto');
            ga('send', 'pageview');
        </script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="..">PESCA</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li >
                                <a href="..">Home</a>
                            </li>
                            <li >
                                <a href="../input_format/">Input Format</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Estimation <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../prior/">Estimating genome-wide priors</a>
</li>
                                    
<li class="active">
    <a href="./">Estimating per-SNP posteriors</a>
</li>
                                </ul>
                            </li>
                            <li >
                                <a href="../faq/">FAQ</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../prior/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../faq/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li>
                                <a href="https://github.com/huwenboshi/pesca"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#estimating-posterior-probabilities-of-each-snp-being-population-specific-and-shared">Estimating posterior probabilities of each SNP being population-specific and shared</a></li>
            <li><a href="#recommended-pipeline">Recommended pipeline</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="estimating-posterior-probabilities-of-each-snp-being-population-specific-and-shared">Estimating posterior probabilities of each SNP being population-specific and shared</h1>
<p>This page describes the pipeline to estimate the posterior probabilities of a
SNP being population-specific and shared.</p>
<h3 id="recommended-pipeline">Recommended pipeline</h3>
<h4 id="step-0-estimate-genome-wide-prior-probabilities">Step 0: estimate genome-wide prior probabilities</h4>
<p>This step should be performend following steps described
<a href="https://huwenboshi.github.io/pesca/prior/">here</a>.</p>
<h4 id="step-1-estimate-per-snp-posterior-probabilities">Step 1: estimate per-SNP posterior probabilities</h4>
<p>PESCA estimates per-SNP posterior probabilities of each SNP beling
population-specific and shared via the following command. This step could be
run for each chromosome in parallel.</p>
<div class="codehilite"><pre><span></span><code><span class="o">&lt;</span><span class="nv">directory</span> <span class="nv">to</span> <span class="nv">pesca</span><span class="o">&gt;/</span><span class="nv">pesca</span> \
    <span class="o">--</span><span class="nv">mode</span> <span class="nv">post</span> \
    <span class="o">--</span><span class="nv">f00</span> <span class="o">&lt;</span><span class="nv">MVB</span> <span class="nv">f00</span> <span class="nv">parameter</span> <span class="nv">from</span> <span class="nv">step</span> <span class="mi">0</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">f01</span> <span class="o">&lt;</span><span class="nv">MVB</span> <span class="nv">f00</span> <span class="nv">parameter</span> <span class="nv">from</span> <span class="nv">step</span> <span class="mi">0</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">f10</span> <span class="o">&lt;</span><span class="nv">MVB</span> <span class="nv">f00</span> <span class="nv">parameter</span> <span class="nv">from</span> <span class="nv">step</span> <span class="mi">0</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">f11</span> <span class="o">&lt;</span><span class="nv">MVB</span> <span class="nv">f00</span> <span class="nv">parameter</span> <span class="nv">from</span> <span class="nv">step</span> <span class="mi">0</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">zscore1</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">GWAS</span> <span class="nv">summary</span> <span class="nv">statistics</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">zscore2</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">GWAS</span> <span class="nv">summary</span> <span class="nv">statistics</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">2</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">ld1</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">LD</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">ld2</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">LD</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">2</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">nburn</span> <span class="o">&lt;</span><span class="nv">number</span> <span class="nv">of</span> <span class="nv">MCMC</span> <span class="nv">burn</span><span class="o">-</span><span class="nv">ins</span>, <span class="mi">5000</span> <span class="nv">by</span> <span class="nv">default</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">nsample</span> <span class="o">&lt;</span><span class="nv">number</span> <span class="nv">of</span> <span class="nv">MCMC</span> <span class="nv">samples</span>, <span class="mi">5000</span> <span class="nv">by</span> <span class="nv">default</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">lambda</span> <span class="o">&lt;</span><span class="nv">shrinkage</span> <span class="nv">parameter</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">sigmasq1</span> <span class="o">&lt;</span><span class="nv">sample</span> <span class="nv">size</span> <span class="nv">times</span> <span class="nv">heritability</span> <span class="nv">in</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">sigmasq2</span> <span class="o">&lt;</span><span class="nv">sample</span> <span class="nv">size</span> <span class="nv">times</span> <span class="nv">heritability</span> <span class="nv">in</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">totnsnp</span> <span class="o">&lt;</span><span class="nv">total</span> <span class="nv">number</span> <span class="nv">of</span> <span class="nv">SNPs</span> <span class="nv">across</span> <span class="nv">all</span> <span class="nv">chromosomes</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">max_iter</span> <span class="o">&lt;</span><span class="nv">number</span> <span class="nv">of</span> <span class="nv">independent</span> <span class="nv">MCMC</span> <span class="nv">chains</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">out</span> <span class="o">&lt;</span><span class="nv">output</span> <span class="nv">file</span> <span class="nv">name</span><span class="o">&gt;</span><span class="nv">_chr</span><span class="o">&lt;</span><span class="nv">chromosome</span> #<span class="o">&gt;</span>
</code></pre></div>


<p>Here are the meaning of the flags:</p>
<ul>
<li>
<p><code>--mode</code> tells PESCA whether to estimate genome-wide prior or per-SNP
posterior probabilities. There are two options for this flag <code>fit</code> and <code>post</code>.
<code>post</code> tells PESCA to estimate posteriors. For estimating priors see 
<a href="https://huwenboshi.github.io/pesca/priors/">here</a>.</p>
</li>
<li>
<p><code>f00</code>, <code>f01</code>, <code>f10</code>, <code>f11</code> are multivariate Bernoulli (MVB) model parameters.
This estimates should be obtained from step 0.</p>
</li>
<li>
<p><code>--zscore1</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 1, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--zscore2</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 2, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--ld1</code> specifies a text file containing a list of paths to LD matrices
for population 1, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore1</code> flag.</p>
</li>
<li>
<p><code>--ld2</code> specifies a text file containing a list of paths to LD matrices
for population 2, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore2</code> flag.</p>
</li>
<li>
<p><code>--nburn</code> specifies the number of burn-ins for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--nsample</code> specifies the number of samples for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--sigmasq1</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 1 multiplied by sample size of the GWAS in population 1.</p>
</li>
<li>
<p><code>--sigmasq2</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 2 multiplied by sample size of the GWAS in population 2.</p>
</li>
<li>
<p><code>--totnsnp</code> specifies total number of SNPs across all chromosomes.</p>
</li>
</ul>
<div style="background-color:rgba(230, 230, 250, 1.0);">
( <b>Note</b>: This is the number of SNPs after intersecting with the
reference panels, which have been MAF filtered and LD pruned. )
</div>

<ul>
<li>
<p><code>--max_iter</code> specifies the number of independent MCMC chains to estimate
posteriors. PESCA uses the average posteriors as the final posteriors. We
recommend setting this value to 20.</p>
</li>
<li>
<p><code>--out</code> specifies the output file name.</p>
</li>
</ul>
<p>The following is an example output, <code>&lt;output file name&gt;_chr&lt;chromosome #&gt;.txt</code>.</p>
<div class="codehilite"><pre><span></span><code>SNP          BP          A0_GWAS1   A1_GWAS1    Z_GWAS1     A0_GWAS2  A1_GWAS2   Z_GWAS2     P_GWAS1_ONLY    P_GWAS2_ONLY    P_BOTH
rs12567000   101242593   C          T           -0.669797   C         T          -1.20845    0.00013         0               0.0736
rs12404364   101255697   T          C           1.40208     T         C          1.45765     0               0.00016         0.10239
rs12043207   101302813   A          G           -0.720678   A         G          0.356864    0               0               0.05533
rs7541397    101322866   G          A           -2.03966    G         A          -3.05884    0.01002         0.00356         0.4613
rs3737581    101341083   G          A           -0.879881   G         A          0.411767    0.0005          0               0.08021
...          ...         ...        ...         ...         ...       ...        ...         ...             ...             ...
</code></pre></div>


<ul>
<li>
<p><code>SNP</code> is the ID of the SNP.</p>
</li>
<li>
<p><code>BP</code> is the base pair position of the SNP.</p>
</li>
<li>
<p><code>A0_GWAS1</code> is the non-effect allele of GWAS in population 1.</p>
</li>
<li>
<p><code>A1_GWAS1</code> is the effect allele of GWAS in population 1.</p>
</li>
<li>
<p><code>Z_GWAS1</code> is the Z-score of GWAS in population 1.</p>
</li>
<li>
<p><code>A0_GWAS2</code> is the non-effect allele of GWAS in population 2.</p>
</li>
<li>
<p><code>A1_GWAS2</code> is the effect allele of GWAS in population 2.</p>
</li>
<li>
<p><code>Z_GWAS2</code> is the Z-score of GWAS in population 2.</p>
</li>
<li>
<p><code>P_GWAS1_ONLY</code> is the posterior probability that the SNP is causal only in
population 1.</p>
</li>
<li>
<p><code>P_GWAS2_ONLY</code> is the posterior probability that the SNP is causal only in
population 2.</p>
</li>
<li>
<p><code>P_BOTH</code> is the posterior probability that the SNP is causal in both
populations.</p>
</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>Copyright &copy; 2020 - 2025 Huwenbo Shi</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>
        <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" defer></script>
        <script src="../assets/mathjaxhelper.js" defer></script>
        <script src="../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
