<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Huwenbo Shi">
        <link rel="canonical" href="https://huwenboshi.github.io/pesca/prior/">
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Estimating genome-wide priors - PESCA</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-XXXXXXXX-X', 'auto');
            ga('send', 'pageview');
        </script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href="..">PESCA</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="../input_format/" class="nav-link">Input Format</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Estimation <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="./" class="dropdown-item active">Estimating genome-wide priors</a>
</li>
                                    
<li>
    <a href="../posterior/" class="dropdown-item">Estimating per-SNP posteriors</a>
</li>
                                </ul>
                            </li>
                            <li class="navitem">
                                <a href="../faq/" class="nav-link">FAQ</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../input_format/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../posterior/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/huwenboshi/pesca" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#estimating-genome-wide-proportion-of-population-specific-and-shared-causal-variants" class="nav-link">Estimating genome-wide proportion of population-specific and shared causal variants</a>
              <ul class="nav flex-column">
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="estimating-genome-wide-proportion-of-population-specific-and-shared-causal-variants">Estimating genome-wide proportion of population-specific and shared causal variants</h1>
<p>This page describes the pipeline to estimate the genome-wide prior
probabilities of a SNP to be causal in one population (i.e.
population-specific) or in both populations (i.e. shared).</p>
<h3 id="step-0-estimate-genome-wide-heritability-of-the-trait">Step 0: estimate genome-wide heritability of the trait</h3>
<p>This step can be performend using <a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">LDSC</a>.</p>
<div style="background-color:rgba(230, 230, 250, 1.0);">
( <b>Note</b>: This step should be performed using all SNPs without any
MAF filtering or LD pruning, except for those required by LDSC. )
</div>

<h3 id="step-1-extract-summary-statistics-and-ld-of-each-region-for-each-chromosome">Step 1: extract summary statistics and LD of each region for each chromosome</h3>
<p>In this step, one</p>
<ul>
<li>
<p>Intersects the GWAS summary statistics with the reference panel. We provide
reference panels in PLINK format for East Asians and Europeans
<a href="https://ucla.box.com/s/o3k9nhosmtjgjufvjjfcshbjynkko8j5">here</a>. This reference
panel contains LD pruned SNPs (\(R^2 &gt; 0.95\)) with MAF greater than 5% in
both populations.</p>
</li>
<li>
<p>Splits GWAS summary statistics data into regions for each chromosome. Format
of the GWAS summary statistics data used by PESCA can be found
<a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>.</p>
</li>
<li>
<p>Computes the corresponding LD matrix of each region. We provide region
definitions that are approximately LD-independent across East Asians
and Europeans <a href="https://github.com/huwenboshi/pesca/tree/master/partition">here</a>.
Format of LD matrix can be found <a href="https://huwenboshi.github.io/pesca/input_format/#ld-matrix">here</a>.</p>
</li>
</ul>
<p>We recommend using the following directory organization.</p>
<div class="codehilite"><pre><span></span><code><span class="err">&lt;trait name&gt;/</span>
<span class="err">    chr1/</span>
<span class="err">        chr1_region1_zscore_pop1.txt</span>
<span class="err">        chr1_region1_zscore_pop2.txt</span>
<span class="err">        chr1_region1_ld_pop1.txt</span>
<span class="err">        chr1_region1_ld_pop2.txt</span>
<span class="err">        ...</span>
<span class="err">        chr1_region100_zscore_pop1.txt</span>
<span class="err">        chr1_region100_zscore_pop2.txt</span>
<span class="err">        chr1_region100_ld_pop1.txt</span>
<span class="err">        chr1_region100_ld_pop2.txt</span>
<span class="err">        ...</span>
<span class="err">        chr1_zscore_list_pop1.txt</span>
<span class="err">        chr1_zscore_list_pop2.txt</span>
<span class="err">        chr1_ld_list_pop1.txt</span>
<span class="err">        chr1_ld_list_pop2.txt</span>
<span class="err">    chr2/</span>
<span class="err">        ...</span>
<span class="err">    ...</span>
<span class="err">    chr22/</span>
<span class="err">        ...</span>
</code></pre></div>


<ul>
<li>
<p><code>chr1_region1_zscore_pop1.txt</code> and <code>chr1_region2_zscore_pop2.txt</code> are
the GWAS summary statistics data files in chromosome 1 region 1 for
population 1 and 2 respectively
(see <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format).</p>
</li>
<li>
<p><code>chr1_region1_ld_pop1.txt</code> and <code>chr1_region1_ld_pop2.txt</code>
are the corresponding LD matrices (see <a href="https://huwenboshi.github.io/pesca/input_format/#ld-matrix">here</a>
for format).</p>
</li>
<li>
<p><code>chr1_zscore_list_pop1.txt</code> is a list of paths to all the <code>chr1_region*_zscore_pop1.txt</code>
files, i.e. GWAS summary statistics data in all regions on chromosome 1 for
population 1. <code>chr1_zscore_list_pop2.txt</code> is a list of paths to all the
<code>chr1_region*_zscore_pop2.txt</code> files.</p>
</li>
<li>
<p><code>chr1_ld_list_pop1.txt</code> is a list of paths to all the <code>chr1_region*_ld_pop1.txt</code>
files, i.e. LD matrices for all regions on chromosome 1 for
population 1. <code>chr1_ld_list_pop2.txt</code> is a list of paths to all the
<code>chr1_region*_ld_pop2.txt</code> files.</p>
</li>
</ul>
<h3 id="step-2-run-pesca-for-each-chromosome">Step 2: run PESCA for each chromosome</h3>
<p>PESCA estimates genome-wide proportions of population-specific and shared
causal variants via the following command. This step could be run for each
chromosome in parallel.</p>
<div class="codehilite"><pre><span></span><code><span class="err">&lt;directory to pesca&gt;/pesca \</span>
<span class="err">    --mode fit \</span>
<span class="err">    --zscore1 &lt;list of GWAS summary statistics files for population 1&gt; \</span>
<span class="err">    --zscore2 &lt;list of GWAS summary statistics files for population 2&gt; \</span>
<span class="err">    --ld1 &lt;list of LD files for population 1&gt; \</span>
<span class="err">    --ld2 &lt;list of LD files for population 2&gt; \</span>
<span class="err">    --nburn &lt;number of MCMC burn-ins, 5000 by default&gt; \</span>
<span class="err">    --nsample &lt;number of MCMC samples, 5000 by default&gt; \</span>
<span class="err">    --lambda &lt;shrinkage parameter, 0.0001 by default&gt; \</span>
<span class="err">    --sigmasq1 &lt;sample size times heritability in population 1&gt; \</span>
<span class="err">    --sigmasq2 &lt;sample size times heritability in population 1&gt; \</span>
<span class="err">    --totnsnp &lt;total number of SNPs across all chromosomes&gt; \</span>
<span class="err">    --max_iter &lt;maximum number of EM iterations&gt; \</span>
<span class="err">    --out &lt;output file name&gt;_chr&lt;chromosome #&gt;</span>
</code></pre></div>


<p>Here are the meaning of the flags:</p>
<ul>
<li>
<p><code>--mode</code> tells PESCA whether to estimate genome-wide prior or per-SNP
posterior probabilities. There are two options for this flag <code>fit</code> and <code>post</code>.
<code>fit</code> tells PESCA to estimate priors. For estimating per-SNP posterior see
<a href="https://huwenboshi.github.io/pesca/posterior/">here</a>.</p>
</li>
<li>
<p><code>--zscore1</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 1, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--zscore2</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 2, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--ld1</code> specifies a text file containing a list of paths to LD matrices
for population 1, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore1</code> flag.</p>
</li>
<li>
<p><code>--ld2</code> specifies a text file containing a list of paths to LD matrices
for population 2, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore2</code> flag.</p>
</li>
<li>
<p><code>--nburn</code> specifies the number of burn-ins for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--nsample</code> specifies the number of samples for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--sigmasq1</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 1 multiplied by sample size of the GWAS in population 1.</p>
</li>
<li>
<p><code>--sigmasq2</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 2 multiplied by sample size of the GWAS in population 2.</p>
</li>
<li>
<p><code>--totnsnp</code> specifies total number of SNPs across all chromosomes.</p>
</li>
</ul>
<div style="background-color:rgba(230, 230, 250, 1.0);">
( <b>Note</b>: This is the number of SNPs after intersecting with the
reference panels, which have been MAF filtered and LD pruned. )
</div>

<ul>
<li>
<p><code>--max_iter</code> specifies maximum number of EM iterations. The default is 200.</p>
</li>
<li>
<p><code>--out</code> specifies the output file name.</p>
</li>
</ul>
<p>The following is an example output, <code>&lt;output file name&gt;_chr&lt;chromosome #&gt;.log</code>.</p>
<div class="codehilite"><pre><span></span><code>iter    nsnp    q00     q01     q10     q11      f00     f01       f10       f11
0       9960    9389.8  190.067 190.067 190.067  0       -3.9      -3.9      3.9
1       9960    9433.58 177.374 184.488 164.561  0       -3.97377  -3.93445  3.85947
2       9960    9466.98 164.664 187.16  141.2    0       -4.05166  -3.9236   3.76987
3       9960    9478.17 164.541 173.529 143.756  0       -4.05359  -4.0004   3.86536
4       9960    9502.39 164.518 157.391 135.704  0       -4.05628  -4.10056  3.90802
5       9960    9524.52 154.285 154.576 126.616  0       -4.12283  -4.12094  3.9233
...     ...     ...     ...     ...     ...      ...     ...       ...       ...
</code></pre></div>


<ul>
<li>
<p><code>iter</code> is the iteration number of the EM algorithm.</p>
</li>
<li>
<p><code>nsnp</code> is the total number of SNPs.</p>
</li>
<li>
<p><code>q00</code> is the number of SNPs that are causal in neither population.</p>
</li>
<li>
<p><code>q01</code> is the number of SNPs that are causal in population 1.</p>
</li>
<li>
<p><code>q10</code> is the number of SNPs that are causal in population 2.</p>
</li>
<li>
<p><code>q11</code> is the number of SNPs that are causal in both populations.</p>
</li>
<li>
<p><code>f00</code>, <code>f01</code>, and <code>f10</code>, and <code>f11</code> are parameters of the multivariate Bernoulli
(MVB) model.</p>
</li>
</ul>
<p>We use results from the last EM iteration as the final estimates.</p>
<h3 id="step-3-aggregate-results-from-all-chromosomes">Step 3: aggregate results from all chromosomes</h3>
<p>After step 2, there should be one <code>&lt;output file name&gt;_chr&lt;chromosome #&gt;.log</code>
for each chromosome. To aggregate results from step 2 across chromosomes, we
provide <a href="https://github.com/huwenboshi/pesca/blob/master/misc/parse_prior.py">parse_prior.py</a>.
This script uses the average of the last 50 iterations of the log file to
estimate genome-wide number of null SNPs (not causal in both populations),
population 1 specific causal SNPs, population 2 specific causal SNPs, and 
shared causal SNPs. The script can be run as follows.</p>
<div class="codehilite"><pre><span></span><code><span class="err">python3 &lt;path to script&gt;/parse_prior.py \</span>
<span class="err">    --prefix &lt;path to results from step 2&gt;/&lt;output file name&gt;_chr</span>
</code></pre></div>


<p>The following is an example output.</p>
<div class="codehilite"><pre><span></span><code>number of SNPs
q00 235640.05
q01 275.53
q10 376.20
q11 21838.23

MVB parameters
f00 0.0000
f01 -6.7514
f10 -6.4399
f11 10.8127
</code></pre></div>


<p>The MVB parameters will be used for estimating the
<a href="https://huwenboshi.github.io/pesca/posterior/">posterior probabilities</a>.</p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>Copyright &copy; 2020 - 2025 Huwenbo Shi</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>
        <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" defer></script>
        <script src="../assets/mathjaxhelper.js" defer></script>
        <script src="../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
