<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Huwenbo Shi">
        <link rel="canonical" href="https://huwenboshi.github.io/pesca/prior/">
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Estimating genome-wide priors - PESCA</title>
        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-XXXXXXXX-X', 'auto');
            ga('send', 'pageview');
        </script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="..">PESCA</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li >
                                <a href="..">Home</a>
                            </li>
                            <li >
                                <a href="../input_format/">Input Format</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Estimation <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li class="active">
    <a href="./">Estimating genome-wide priors</a>
</li>
                                    
<li >
    <a href="../posterior/">Estimating per-SNP posteriors</a>
</li>
                                </ul>
                            </li>
                            <li >
                                <a href="../faq/">FAQ</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../input_format/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../posterior/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li>
                                <a href="https://github.com/huwenboshi/pesca"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#estimating-genome-wide-proportion-of-population-specific-and-shared-causal-variants">Estimating genome-wide proportion of population-specific and shared causal variants</a></li>
            <li><a href="#recommended-pipeline">Recommended pipeline</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="estimating-genome-wide-proportion-of-population-specific-and-shared-causal-variants">Estimating genome-wide proportion of population-specific and shared causal variants</h1>
<p>This page describes the pipeline to estimate the genome-wide prior
probabilities of a SNP to be causal in one population (i.e.
population-specific) or in both populations (i.e. shared).</p>
<h3 id="recommended-pipeline">Recommended pipeline</h3>
<h4 id="step-1-estimate-genome-wide-heritability-of-the-trait">Step 1: estimate genome-wide heritability of the trait</h4>
<p>This step can be performend using <a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">LDSC</a>.</p>
<div style="background-color:rgba(230, 230, 250, 1.0);">
( <b>Note</b>: This step should be performed using all SNPs without any
MAF filtering or LD-pruning, except for those required by LDSC. )
</div>

<h4 id="step-2-extract-summary-statistics-and-ld-of-each-region-for-each-chromosome">Step 2: extract summary statistics and LD of each region for each chromosome</h4>
<p>In this step, one</p>
<ul>
<li>
<p>Intersects the GWAS summary statistics with the reference panel. We provide
reference panels in PLINK format for East Asians and Europeans
<a href="https://ucla.box.com/s/o3k9nhosmtjgjufvjjfcshbjynkko8j5">here</a>. This reference
panel contains LD pruned SNPs (\(R^2 &gt; 0.95\)) with MAF greater than 5% in
both populations.</p>
</li>
<li>
<p>Splits GWAS summary statistics data into regions for each chromosome. Format
of the GWAS summary statistics data used by PESCA can be found
<a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>.</p>
</li>
<li>
<p>Computes the corresponding LD matrix of each region. We provide region
definitions that are approximately LD-independent across East Asians
and Europeans <a href="https://github.com/huwenboshi/pesca/tree/master/partition">here</a>.
Format of LD matrix can be found <a href="https://huwenboshi.github.io/pesca/input_format/#ld-matrix">here</a>.</p>
</li>
</ul>
<p>We recommend using the following directory organization.</p>
<div class="codehilite"><pre><span></span><code><span class="o">&lt;</span><span class="n">trait</span> <span class="n">name</span><span class="o">&gt;/</span>
    <span class="n">chr1</span><span class="o">/</span>
        <span class="n">chr1_region1_zscore_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region1_zscore_pop2</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region1_ld_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region1_ld_pop2</span><span class="p">.</span><span class="n">txt</span>
        <span class="p">...</span>
        <span class="n">chr1_region100_zscore_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region100_zscore_pop2</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region100_ld_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_region100_ld_pop2</span><span class="p">.</span><span class="n">txt</span>
        <span class="p">...</span>
        <span class="n">chr1_zscore_list_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_zscore_list_pop2</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_ld_list_pop1</span><span class="p">.</span><span class="n">txt</span>
        <span class="n">chr1_ld_list_pop2</span><span class="p">.</span><span class="n">txt</span>
    <span class="n">chr2</span><span class="o">/</span>
        <span class="p">...</span>
    <span class="p">...</span>
    <span class="n">chr22</span><span class="o">/</span>
        <span class="p">...</span>
</code></pre></div>


<ul>
<li>
<p><code>chr1_region1_zscore_pop1.txt</code> and <code>chr1_region2_zscore_pop2.txt</code> are
the GWAS summary statistics data files in chromosome 1 region 1 for
population 1 and 2 respectively
(see <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format).</p>
</li>
<li>
<p><code>chr1_region1_ld_pop1.txt</code> and <code>chr1_region1_ld_pop2.txt</code>
are the corresponding LD matrices (see <a href="https://huwenboshi.github.io/pesca/input_format/#ld-matrix">here</a>
for format).</p>
</li>
<li>
<p><code>chr1_zscore_list_pop1.txt</code> is a list of paths to all the <code>chr1_region*_zscore_pop1.txt</code>
files, i.e. GWAS summary statistics data in all regions on chromosome 1 for
population 1. <code>chr1_zscore_list_pop2.txt</code> is a list of paths to all the
<code>chr1_region*_zscore_pop2.txt</code> files.</p>
</li>
<li>
<p><code>chr1_ld_list_pop1.txt</code> is a list of paths to all the <code>chr1_region*_ld_pop1.txt</code>
files, i.e. LD matrices for all regions on chromosome 1 for
population 1. <code>chr1_ld_list_pop2.txt</code> is a list of paths to all the
<code>chr1_region*_ld_pop2.txt</code> files.</p>
</li>
</ul>
<h4 id="step-3-run-pesca-for-each-chromosome">Step 3: run PESCA for each chromosome</h4>
<p>Command for running PESCA</p>
<p>PESCA estimates genome-wide proportions of population-specific and shared
causal variants via the following command.</p>
<div class="codehilite"><pre><span></span><code><span class="o">&lt;</span><span class="nv">directory</span> <span class="nv">to</span> <span class="nv">pesca</span><span class="o">&gt;/</span><span class="nv">pesca</span> \
    <span class="o">--</span><span class="nv">mode</span> <span class="nv">fit</span> \
    <span class="o">--</span><span class="nv">zscore1</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">GWAS</span> <span class="nv">summary</span> <span class="nv">statistics</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">zscore2</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">GWAS</span> <span class="nv">summary</span> <span class="nv">statistics</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">2</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">ld1</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">LD</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">1</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">ld2</span> <span class="o">&lt;</span><span class="nv">list</span> <span class="nv">of</span> <span class="nv">LD</span> <span class="nv">files</span> <span class="k">for</span> <span class="nv">population</span> <span class="mi">2</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">nburn</span> <span class="o">&lt;</span><span class="nv">number</span> <span class="nv">of</span> <span class="nv">MCMC</span> <span class="nv">burn</span><span class="o">-</span><span class="nv">ins</span>, <span class="mi">5000</span> <span class="nv">by</span> <span class="nv">default</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">nsample</span> <span class="o">&lt;</span><span class="nv">number</span> <span class="nv">of</span> <span class="nv">MCMC</span> <span class="nv">samples</span>, <span class="mi">5000</span> <span class="nv">by</span> <span class="nv">default</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">lambda</span> <span class="o">&lt;</span><span class="nv">shrinkage</span> <span class="nv">parameter</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">sigmasq1</span> $<span class="nv">sigmasq1</span> \
    <span class="o">--</span><span class="nv">sigmasq2</span>  $<span class="nv">sigmasq2</span> \
    <span class="o">--</span><span class="nv">totnsnp</span> <span class="o">&lt;</span><span class="nv">total</span> <span class="nv">number</span> <span class="nv">of</span> <span class="nv">SNPs</span> <span class="nv">across</span> <span class="nv">all</span> <span class="nv">chromosomes</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">max_iter</span> <span class="o">&lt;</span><span class="nv">maximum</span> <span class="nv">number</span> <span class="nv">of</span> <span class="nv">EM</span> <span class="nv">iterations</span><span class="o">&gt;</span> \
    <span class="o">--</span><span class="nv">out</span> <span class="o">&lt;</span><span class="nv">output</span> <span class="nv">file</span> <span class="nv">name</span><span class="o">&gt;</span>
</code></pre></div>


<p>Here are the meaning of the flags:</p>
<ul>
<li>
<p><code>--model</code> tells PESCA whether to estimate genome-wide prior or per-SNP
posterior probabilities. There are two options for this flag <code>fit</code> and <code>post</code>.
For estimating per-SNP posterior see
<a href="https://huwenboshi.github.io/pesca/posterior/">here</a>.</p>
</li>
<li>
<p><code>--zscore1</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 1, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--zscore2</code> specifies a text file containing a list of paths to GWAS
summary statistics for population 2, one path per line. Typically, there
should be one such text file for each chromosome. And each line in the text
file corresponds to GWAS summary statistics of one region on that chromosome.
See <a href="https://huwenboshi.github.io/pesca/input_format/#gwas-summary-statistics-data">here</a>
for format of GWAS summary statistics data for PESCA.</p>
</li>
<li>
<p><code>--ld1</code> specifies a text file containing a list of paths to LD matrices
for population 1, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore1</code> flag.</p>
</li>
<li>
<p><code>--ld2</code> specifies a text file containing a list of paths to LD matrices
for population 2, one path per line. Typically, there should be one such
text file for each chromosome. And each line in the text file corresponds
to LD matrix of one region on that chromosome. Additionally, each line
in the text file should correspond to the same region listed by the
<code>--zscore2</code> flag.</p>
</li>
<li>
<p><code>--nburn</code> specifies the number of burn-ins for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--nsample</code> specifies the number of samples for the MCMC. The default is 5000.</p>
</li>
<li>
<p><code>--sigmasq1</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 1 multiplied by sample size of the GWAS in population 1.</p>
</li>
<li>
<p><code>--sigmasq2</code> specifies genome-wide SNP-heritability (e.g. estimated by
LDSC) for population 2 multiplied by sample size of the GWAS in population 2.</p>
</li>
<li>
<p><code>--totnsnp</code> specifies total number of SNPs across all chromosomes.</p>
</li>
<li>
<p><code>--max_iter</code> specifies maximum number of EM iterations. The default is 200.</p>
</li>
<li>
<p><code>--out</code> specifies the output file name.</p>
</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>Copyright &copy; 2020 - 2025 Huwenbo Shi</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>
        <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" defer></script>
        <script src="../assets/mathjaxhelper.js" defer></script>
        <script src="../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
